### PR TITLE
feat(admin): return `config_dump` response in the same format as envoy admin 

### PR DESCRIPTION
**Pull request by lukidzi**
### Checklist prior to review

- [ ] [Link to relevant issue][1] as well as docs and UI issues -- fix: https://github.com/kumahq/kuma/issues/8510
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/policies/core/jsonpatch/merge_test.go [hunk 1]
```diff
@@ -67,9 +67,17 @@ var _ = Describe("Json Patch merge", func() {
 		}
 
 		// when
-		_, err := jsonpatch.MergeJsonPatchAny(util_proto.MustMarshalAny(hcm), patches)
+		hcmAny := util_proto.MustMarshalAny(hcm)
+		mergedHcmAny, err := jsonpatch.MergeJsonPatchAny(hcmAny, patches)
```
Comment (slonka) [inline pkg/plugins/policies/core/jsonpatch/merge_test.go:71]: how does it create camelCase?

**Review by slonka — APPROVED** [review]
if it does not screw up anything else, and  you're sure in your test you're creating `camelCase`

File: pkg/plugins/policies/core/jsonpatch/merge_test.go (hunk 1 above)
Comment (lukidzi) [reply pkg/plugins/policies/core/jsonpatch/merge_test.go:71]: I've used a library but if we think it is error prone we could try do regex and match->replace https://github.com/gobeam/stringy/blob/master/stringy.go#L112

**Comment by lahabana** [comment]
How come the dump and the values do not match?

**Comment by lukidzi** [comment]
> How come the dump and the values do not match?

I did some debugging, it is caused by not setting `OrigName` to true https://github.com/kumahq/kuma/blob/master/pkg/util/proto/proto.go#L44
```golang
marshaler := &jsonpb.Marshaler{
		OrigName: true,
	}
```

if we change it to this we are going to have proto fields name (like in a config_dump)

`MeshProxyPatch` allows the creation of `jsonPatch` and accepts JSON formatted fields `camelCase`; if we change the dump, a user might have to rewrite them. The same is true with the `value` field.

**Comment by lukidzi** [comment]
I think this issue might not be the case if we agree that what presents UI is true, but if we decide that we should stick to the same format as Envoy(which would be the best for cross docs reading) then we need to rewrite a lot of things and we need to keep backward compatibility for old patches in JSON format

**Review by jakubdyszkiewicz — APPROVED** [review]

File: pkg/envoy/admin/testdata/golden.full_config.json [hunk 2]
```diff
@@ -7,41 +7,41 @@
           "id": "default.backend-1",
```
Comment (lukidzi) [inline pkg/envoy/admin/testdata/golden.full_config.json:7]: When compared with the source file it has only one change:
image

File: pkg/envoy/admin/sanitize.go [hunk 3]
```diff
@@ -1,35 +1,42 @@
...
+	"strings"
 
-func Sanitize(configDump *envoy_admin_v3.ConfigDump) error {
-	for _, config := range configDump.Configs {
+	envoy_admin_v3 "github.com/envoyproxy/go-control-plane/envoy/admin/v3"
+
+	util_proto "github.com/kumahq/kuma/pkg/util/proto"
+)
+
+func Sanitize(configDump []byte) ([]byte, error) {
```
Comment (lukidzi) [inline pkg/envoy/admin/sanitize.go:11]: @lahabana I've changed to the way you mentioned in the issue

File: api/common/v1alpha1/jsonpatch.go [hunk 4]
```diff
@@ -5,7 +5,7 @@ import (
 	"fmt"
 	"strconv"
 
-	"github.com/evanphx/json-patch/v5"
+	jsonpatch "github.com/evanphx/json-patch/v5"
```
Comment (lahabana) [inline api/common/v1alpha1/jsonpatch.go:8]: Feels like it's left over
```suggestion
                    "github.com/evanphx/json-patch/v5"
```

**Pull request merged**